internal/jimm/juju/applicationoffer_test.go

```diff
@@ -287,44 +287,35 @@ func TestGetApplicationOfferConsumeDetails(t *testing.T) {
 	err = openfga.NewUser(uAll, client).SetApplicationOfferAccess(ctx, offer.ResourceTag(), ofganames.ReaderRelation)
 	c.Assert(err, qt.IsNil)
 
-	j := jimmtest.NewJIMM(c, &jimm.Parameters{
```

Comment:
This change was made to support the use of the `NewTestJujuManager` method which returns a JujuManager with a database and openfga client, etc. This test was first creating a DB then creating a bunch of items then setting up the API mock to return the UUID/IDs of the returned items. So I've moved the creation of JIMM above, so that the DB is available, then setup the API mock lower down before the test begins.